skbio/sequence/_feature.py

```diff
@@ -0,0 +1,52 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2013--, scikit-bio development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+# ----------------------------------------------------------------------------
+
+from __future__ import absolute_import, division, print_function
+
+from collections import Mapping
+import numpy as np
+
+
+class Feature(Mapping):
+    '''Store the metadata of a sequence feature.
```

Comment:
If we are going to do that,  might as well hard code location no?
On Feb 11, 2016 11:00 PM, "Zech Xu" <notifications@github.com> wrote:

> In skbio/sequence/_feature.py
> <https://github.com/RNAer/scikit-bio/pull/1#discussion_r52710617>:
>
> > +# ----------------------------------------------------------------------------
> > +# Copyright (c) 2013--, scikit-bio development team.
> > +#
> > +# Distributed under the terms of the Modified BSD License.
> > +#
> > +# The full license is in the file COPYING.txt, distributed with this software.
> > +# ----------------------------------------------------------------------------
> > +
> > +from __future__ import absolute_import, division, print_function
> > +
> > +from collections import Mapping
> > +import numpy as np
> > +
> > +
> > +class Feature(Mapping):
> > +    '''Store the metadata of a sequence feature.
>
> we should enforce the presence of some keys including strand, left (open
> in left end), right (open in right end), type (feature type) and give them
> default value if not set, as these info should be required for every
> feature.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/RNAer/scikit-bio/pull/1/files#r52710617>.
>